### ISSUE TITLE
A list inside an alias on a markdown page repeats the alias indefinitely

### ISSUE THREAD

**Issue by emmenlau** [doxygen/doxygen#10504]
**Describe the bug**
I have the following alias in my `doxygen.cfg`:
```
ALIASES += highlight_green{1}="<div style='border: 2px solid green; background-color: #FFFFEE; margin: 10px 0px 20px 0px; padding: 10px;'>\1</div>"
```

Generally it works nicely on markdown pages, so I can i.e. write:
```
@highlight_green{Here is a relevant aspect!}
```

However I'd like to also put lists inside the alias, like this:
```
@highlight_green{
 - Consider this aspect: A
 - Also consider the other aspect: B
}
```

When I do that, the resulting page shows all text following the highlight also with the same highlight!

**Expected behavior**
Only the list inside the alias should be rendered with the alias formatting.

**Screenshots**
Here is the screenshot of the problem:
![image](https://github.com/doxygen/doxygen/assets/1098087/53ceb907-a298-47dd-81fa-e0227d66e89b)

**To Reproduce**
Here is the full example: [doxygen-issue-10504.zip](https://github.com/doxygen/doxygen/files/13753539/doxygen-issue-10504.zip)

**Version**
1.9.8 (https://github.com/doxygen/doxygen/commit/c2fe5c3e4986974eb2a97608b24086683502f07f) installed from the official website, running on Ubuntu 22.04 x86_64.

**Comment by albert-github** [comment]
> Here is the full example:

Where or am I too fast?

**Comment by emmenlau** [comment]
> > Here is the full example:
> 
> Where or am I too fast?

Hahaha, you are **really** fast! Thanks! I added the issue number to the zip file, therefore the upload was delayed :-)

**Comment by emmenlau** [comment]
Also, I can work around the problem by using an HTML unordered list, like so:
```
@highlight_green{
<ul>
<li>List entry A.</li>
<li>List entry B.</li>
</ul>
}

This line is not highlighted, which is great!
```

**Comment by albert-github** [comment]
Indeed, the mentioned problem is also noted in the warnings:
```
.../input/frontpage.md:13: warning: found </div> at different nesting level (5) than expected (2)
.../input/frontpage.md:16: warning: end of comment block while expecting command </div>
```
this is something that should be investigated (in the doxygen code).

Note: you don't generated LaTeX but the alias could be a bit more sophisticated / working for other formats as well by changing:
```
ALIASES                = "highlight_green{1}=<div style='border: 2px solid green; background-color: #FFFFEE; margin: 10px 0px 20px 0px; padding: 10px;'>\1</div>"
```
into
```
ALIASES                = "highlight_green{1}=\htmlonly<div style='border: 2px solid green; background-color: #FFFFEE; margin: 10px 0px 20px 0px; padding: 10px;'>\endhtmlonly\1\htmlonly</div>\endhtmlonly"
```

It also sees to it that the original problem disappears.

**Comment by emmenlau** [comment]
> ```
> ALIASES                = "highlight_green{1}=\htmlonly<div style='border: 2px solid green; background-color: #FFFFEE; margin: 10px 0px 20px 0px; padding: 10px;'>\endhtmlonly\1\htmlonly</div>\endhtmlonly"
> ```

Awesome support, thanks a lot @albert-github !!! This issue is then rather low priority, because the workaround is perfectly valid and I can confirm it works for me (out of the box)!

Thanks a lot!

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #10520

**Comment by emmenlau** [comment]
Nice work, thanks @albert-github ! I'll try to test current master soon.

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.11.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).

**Comment by emmenlau** [comment]
Thanks a lot for your great work!!
